CMS-Enterprise/mint-app
# [MINT-3217] Common milestone library

# MINT-3217

## Description


-  Added view for milestone library
- Added component for `<MilestoneCard />`
- Added component and util for `<SuggestedMilestoneToggle />`
- Added unit test

Can add this code to manually add a suggested/added milestone

```
  // TEMP only for demo
  const milestones = useMemo(() => {
    const rawData =
      data?.modelPlan?.mtoMatrix?.commonMilestones ||
      ([] as MilestoneCardType[]);

    const testData = JSON.parse(JSON.stringify(rawData));

    if (testData.length > 0) {
      testData[0].isSuggested = true;
      testData[0].isAdded = true;
    }

    return testData;
  }, [data?.modelPlan?.mtoMatrix?.commonMilestones]);
```


## How to test this change


1. Create model plan and navigate to MTO
2. Click on 'Browse common milestone'
3. Verify table works with pagination, search, and page size
4. Verify matched figma

Why is this suggested is still in progress pending BE
Add to matrix still pending BE
About this milestone still pending


TODO: Add unit test/file for `SuggestedMilestoneToggle` once the BE work is complete

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
